### PR TITLE
[ENH] add readme to default data locations

### LINKED ISSUE #1803: Add a README.txt file in the nilearn_data folder
When we create the nilearn_data folder, we should add a file that says what it is, and that it can be safely deleted.

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #1803

Changes proposed in this pull request:

- add readme in default data location to let user know it is sage to remove this folder

### CODE REVIEW

File: nilearn/datasets/utils.py [hunk 1]
```diff
@@ -316,6 +317,23 @@ def _get_dataset_dir(
...
 
+def _add_readme_to_default_data_locations(verbose):
+    for d in get_data_dirs():
+        file = Path(d) / "README.md"
+        if file.parent.exists() and not file.exists():
+            with open(file, "w") as f:
+                f.write(
+                    """# Nilearn data folder
+
+This directory is used by nilearn to store datasets
```
Comment (bthirion) [inline nilearn/datasets/utils.py:328]:
```suggestion
This directory is used by Nilearn to store datasets
```

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

**Comment by Remi-Gau** [comment]
@bthirion @ymzayek 
Does this need a test? I am leaning that this is a bit of an overkill but happy to add one if you feel otherwise.

**Review by ymzayek — COMMENTED** [review]
LGTM! Maybe not necessary but it would be good to just test that the file is created in the right location.

File: nilearn/datasets/tests/test_utils.py [hunk 2]
```diff
@@ -16,9 +16,7 @@
 import pytest
 import requests
 
-from nilearn import datasets
 from nilearn.datasets import utils
-from nilearn.datasets.utils import _get_dataset_descr
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_utils.py:21]: unrelated but simplified the imports a bit

File: nilearn/datasets/tests/test_utils.py [hunk 3]
```diff
@@ -121,7 +119,13 @@ def test_get_dataset_dir(tmp_path):
...
         "directories, but",
     ):
-        datasets.utils._get_dataset_dir("test", test_file, verbose=0)
+        utils._get_dataset_dir("test", test_file, verbose=0)
+
+
+def test_add_readme_to_default_data_locations(tmp_path):
+    assert not (tmp_path / "README.md").exists()
+    utils._get_dataset_dir(dataset_name="test", verbose=0, data_dir=tmp_path)
+    assert (tmp_path / "README.md").exists()
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_utils.py:128]: simple test added

**Pull request merged**